plugins/obs-filters/data/cross_stitching_filter_gl.effect

```diff
@@ -0,0 +1,143 @@
+/****************************************************************************
+Copyright (C) 2017 by c3r1c3 <c3r1c3@nevermindonline.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*****************************************************************************/
+
+/* From the "Cross-Stitching" filter from
+ * http://www.geeks3d.com/20110408/cross-stitching-post-processing-shader-glsl
+  -filter-geexlab-pixel-bender/
+ * Adapted for OBS by c3r1c3
+ */
+
+/* Standard OBS->GPU uploads. */
+uniform float4x4 ViewProj;
+uniform texture2d image;
+
+/* Filter variables from OBS. */
+uniform float  stitch_gap;
+uniform float  negative_bias;
+uniform float4 stitch_color;
+uniform bool   invert;
+uniform bool   greyscale;
+
+/*
+ * Here we set how the base texture will be sampled/read/interpreted.
+ * Below is what you would normally use in OBS. Why?
+ * Linear filtering 'messes' with the image the least.
+ */
+sampler_state textureSampler {
+	Filter   = Linear;
+	AddressU = Clamp;
+	AddressV = Clamp;
+};
+
+/*
+ * Here we create the struct to hold the current position and texture info
+ * of the vertex in question.
+ */
+struct VertData {
+	float4 pos : POSITION;
+	float2 uv  : TEXCOORD0;
+};
+
+/*
+ * Now we transform the vertices to world space (and the
+ * textures to go with it). In OBS you'll usually want to
+ * use the code below, unless you're trying to create a
+ * '3D'-esqe effect.
+ */
+VertData VSDefault(VertData vert_in)
+{
+	VertData vert_out;
+	vert_out.pos = mul(float4(vert_in.pos.xyz, 1.0), ViewProj);
+	vert_out.uv = vert_in.uv;
+	return vert_out;
+}
+
+/*
+ * Lastly we have the function that transforms the texture into
+ * the "Cross-Stitched" version of itself.
+ */
+float4 PSCrossStitch(VertData vert_in) : TARGET
+{
+	float4 pix_pos;
+	float4 color;
+	float4 transColor;
+	float  luminance;
+	float2 textureSized;
+
+	/* Get current Texture Size */
+	textureSized = textureSize(image,0);
+
+	/* Get the current pixels' color. */
```

Comment:
If this is referring to a single pixel, use the singular possessive "pixel's" with an apostrophe followed by the letter "s".  If it's referring to a group of pixels, disregard this request.